analyses/fusion_filtering/04-project-specific-filtering.Rmd

```diff
@@ -0,0 +1,269 @@
+
+
+
+
+
+
+```
+
+
+```{r}
+# Gene fusion should be in-frame
+# AND
+#
+# 1. Called by at least n callers
+fusion_calls.summary <- fusion_calls %>%
+  dplyr::filter(caller.count >= numCaller) %>%
+  unique() %>%
+  mutate(note=paste0("Called by",numCaller, "callers")) %>%
+  as.data.frame()
+
+# OR
+# 2. Found in at least n samples in each group
+sample.count <- fusion_calls %>%
+  dplyr::filter(Fusion_Type != "other") %>%
+  dplyr::select(FusionName, Sample, !!as.name(group),-Fusion_Type) %>%
+  unique() %>%
+  group_by(FusionName, !!as.name(group)) %>%
+  dplyr::mutate(sample.count = n(),Sample = toString(Sample)) %>%
+  dplyr::filter(sample.count > numSample) %>%
+  unique() %>%
+  mutate(note=paste0("Found in atleast ",numSample, " samples in a group")) %>%
+  as.data.frame()
+
+
+```
+
+
+
+```{r}
+#filter QCGeneFiltered_filtFusion to keep recurrent fusions from above sample.count and fusion_calls.summary
+
+QCGeneFiltered_recFusion<-fusion_calls %>%
+  dplyr::filter(FusionName %in% unique(c(sample.count$FusionName,fusion_calls.summary$FusionName)))
+
+```
+
+
+
+
+
+```{r}
+# remove fusions that are in > numGroup
+group.count <- fusion_calls %>%
+  dplyr::select(FusionName, !!as.name(group)) %>%
+  unique() %>%
+  group_by(FusionName) %>%
+  dplyr::mutate(group.ct = n(),Sample = toString(!!(as.name(group)))) %>%
+  dplyr::filter(group.ct >numGroup)
+
+
+# remove multi-fused genes
+fusion_recurrent5_per_sample <- fusion_calls  %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample, FusionName, Gene1A, Gene1B, Gene2A, Gene2B) %>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = GeneSymbol) %>%
+  # Remove columns without gene symbols
+  dplyr::filter(GeneSymbol != "") %>%
+  dplyr::arrange(Sample, FusionName) %>%
+  # Retain only distinct rows
+  dplyr::distinct() %>%
+  group_by(Sample,GeneSymbol) %>% 
+  dplyr::summarise(Gene.ct = n()) %>% 
+  dplyr::filter(Gene.ct>limitMultiFused) %>%
+  mutate(note=paste0("multfused " ,limitMultiFused, " times per sample"))
+
+
+
+```
+
+
+```{r}
+
+
+outputfolder<-"results"
+setwd(outputfolder)
+# filter QCGeneFiltered_recFusion to remove fusions found in more than 1 group 
+
+QCGeneFiltered_recFusionUniq<-QCGeneFiltered_recFusion %>%
+  dplyr::filter(!FusionName %in% group.count$FusionName) %>%
+  dplyr::filter(!Gene1A %in% fusion_recurrent5_per_sample$GeneSymbol |
+                  !Gene2A %in% fusion_recurrent5_per_sample$GeneSymbol |
+                  !Gene1B %in% fusion_recurrent5_per_sample$GeneSymbol | 
+                  !Gene2B %in% fusion_recurrent5_per_sample$GeneSymbol) %>%
+  unique()
+
+
+
+
+```
+
+
+
+
+
+```{r}
+setwd(outputfolder)
+
+# filter for putative driver genes and mutifused genes per sample
+putative_driver_annotated_fusions <- fusion_calls %>%
+  # dplyr::filter(!Gene1A %in% fusion_recurrent5_per_sample$GeneSymbol |
```

Comment:
I would remove these lines that are commented out since you do this step above.

Reply:
done